src/test/java/SpendingAnalyticsServletTest.java

```diff
@@ -35,6 +35,9 @@
 import org.mockito.MockitoAnnotations;
 
 public final class SpendingAnalyticsServletTest {
+  private static final HashMap<String, Double> EXPECTED_STORE_ANALYTICS = new HashMap<>();
+  private static final HashMap<String, Double> EXPECTED_CATEGORY_ANALYTICS = new HashMap<>();
```

Comment:
I recommend initializing the map here so it's only done once for all the tests. I think the below should work:
`new HashMap(ImmutableMap.of("walmart", 26.12, "contoso", 14.51, "main street restaurant", 29.01))`